### ISSUE TITLE
sendIntrospectionQuery gives ECONNREFUSED with localhost server

### ISSUE THREAD

**Issue by NicholasLYang** [apollographql/reason-apollo#180]
This is a bit of a puzzler (and quite possibly not the fault of this library at all). I'm running my Rust/Juniper GraphQL server on localhost (port 4000) and attempting to run sendIntrospectionQuery. 

A quick aside, the docs aren't really friendly to people who are using npm. Any idea what the equivalent npm command is? I'd be happy to open a PR to add it to the documentation. I'm running `node node_modules/graphql_ppx/sendIntrospectionQuery` if it matters. 

Anyways, the script fails with an ECONNREFUSED and my server doesn't even appear to notice the request. However, when I proxy the server using ngrok, this request goes through. Any idea why this happens?

Thanks,
Nicholas

**Comment by Gregoirevda** [comment]
From npm@5.2, you can use npx
`npx send-introspection-query http://my-api.example.com/api`
otherwise, you need to expand your scripts in `package.json` 

```
javascript
 scripts : {
     send-introspection-query: send-introspection-query http://my-api.example.com/api
 }
```

Can you provide details of the `ECONNREFUSED`? 
Do you have cors enabled on your GraphQL server?

**Comment by NicholasLYang** [comment]
Ah, okay. I get the following error, which doesn't appear to be that useful:
```
Could not send introspection query:  { Error: connect ECONNREFUSED 127.0.0.1:4000
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1113:14)
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 4000 }
```
I do have CORS enabled on my server. Quite honestly I'm leaning towards something weird on my end, because I read the sendIntrospectionQuery code and it looks pretty straightforward.

**Comment by Gregoirevda** [comment]
Can you call it from a client supporting GraphQL like Insomnia?
Strange it only works from ngrock, sure it running in the correct place? `lsof -i tcp:4000`

**Comment by NicholasLYang** [comment]
Yeah, the server is running in the right place. I even did a `kill -9` and restarted it to make sure it wasn't a hidden process that I didn't kill properly. My GraphiQL client works properly too. And actually, my front end app also works too now that I set up the schema via ngrok.

**Comment by idkjs** [comment]
@NicholasLYang you ever figure this out?

**Comment by idkjs** [comment]
Disregard my comments. Typo in url. Had `http://localhost/4000` instead of `http://localhost:4000`

**Comment by Ejhfast** [comment]
@NicholasLYang this issue also drove me a bit crazy today. It seems that Rocket does has some issues serving on localhost: https://stackoverflow.com/a/59668606

For me the following worked:
```npx get-graphql-schema http://[::1]:4000/graphql -j```

Whereas the query I would typically do does not work:
```npx get-graphql-schema http://localhost:4000/graphql -j```

Hope others can benefit from this comment, as this was really surprising and hard to debug. I imagine many people are combining a Rust/Rocket/Juniper backend and Reason/Apollo frontend, so perhaps it is worth calling out this issue somewhere in the documentation.

**Comment by renehauck** [comment]
For my future self and also all other help seekers: If you use the local route ip via `ifconfig` instead of localhost, it worked for me.
